autoflip_calculators/text_detection_calculator.cc

```diff
@@ -172,88 +159,94 @@ void TextDetectionCalculator::DecodeBoundingBoxes(const Mat& scores, const Mat&
   }
 
-  if (cc->Inputs().HasTag("VIDEO")) {
-    cv::Mat frame, blob;
-    frame = mediapipe::formats::MatView(
-        &cc->Inputs().Tag("VIDEO").Get<ImageFrame>());
-    frame_width_ = frame.cols;
-    frame_height_ = frame.rows;
-
-    // Mean subtraction and scalling.
-    blobFromImage(frame, blob, 1.0, Size(frame_width_, frame_height_), 
-                  Scalar(123.68, 116.78, 103.94), true, false);
-    // Detect the text.
-    detector.setInput(blob);
-    detector.forward(outs, outNames);
-
-    cv::Mat scores = outs[0];
-    cv::Mat geometry = outs[1];
-
-    // Decode predicted bounding boxes and corresponding confident scores.
-    std::vector<RotatedRect> boxes;
-    std::vector<float> confidences;
-    DecodeBoundingBoxes(scores, geometry, confidence_threshold, boxes, confidences);
-
-    // Apply non-maximum suppression procedure.
-    std::vector<int> indices;
-    cv::dnn::NMSBoxes(boxes, confidences, confidence_threshold, nms_threshold, indices);
-
-    auto region_set = ::absl::make_unique<DetectionSet>();
-    for (size_t i = 0; i < indices.size(); ++i) {
-      cv::RotatedRect& box = boxes[indices[i]];
-      cv::Rect2f RectBox = box.boundingRect();
-      float text_score = confidences[indices[i]];
-      
-      // Normalize the bounding box.
-      RectBox.x = NormalizeX(RectBox.x);
-      RectBox.y = NormalizeY(RectBox.y);
-      RectBox.width = NormalizeX(RectBox.width);
-      RectBox.height = NormalizeY(RectBox.height);
-      float x = std::max(0.0f, RectBox.x);
-      float y = std::max(0.0f, RectBox.y);
-      float width =
-          std::min(NormalizeX(RectBox.width) - x + RectBox.x, 1 - x);
-      float height = 
-          std::min(RectBox.height - y + RectBox.y, 1 - y);
-
-      // Convert the text bounding box to a region.
-      if (options_.export_text_bounding_box()) {
-        SalientRegion* region = region_set->add_detections();
-        region->mutable_location_normalized()->set_x(x);
-        region->mutable_location_normalized()->set_y(y);
-        region->mutable_location_normalized()->set_width(width);
-        region->mutable_location_normalized()->set_height(height);
-        region->mutable_signal_type()->set_standard(SignalType::TEXT);
-
-        // Score the text based on image cues.
-        if (options_.use_visual_scorer()) {
-          MP_RETURN_IF_ERROR(
-              scorer_->CalculateScore(frame, *region, &text_score));
-        }
-        region->set_score(text_score);
-      }
+  cv::Mat frame;
+  frame = mediapipe::formats::MatView(
+      &cc->Inputs().Tag(kInputVideo).Get<ImageFrame>());
+
+  // Detect the text.
+  cv::Mat scores, geometry;
+  DetectText(frame, detector_, scores, geometry);
+
+  // Decode predicted bounding boxes and corresponding confident scores.
+  std::vector<cv::RotatedRect> boxes;
+  std::vector<float> confidences;
+  DecodeBoundingBoxes(scores, geometry, kConfidence_threshold_, boxes, confidences);
```

Comment:
Do you think it would be better to put the kConfidence_threshold in the options proto?  That way it can be directly adjusted in the mediapipe graph without recompiling.